client/src/components/dashboard/Dashboard.js

```diff
@@ -3,52 +3,81 @@
  */
 import React, { useState } from 'react'
 import { Container, Button, Modal, Image, CardGroup, Card, Form } from "react-bootstrap"
+import { Link } from 'react-router-dom';
 
 /** Redux **/
 import { useDispatch, useSelector } from 'react-redux';
 import { createWebsite } from '../../actions/website';
 
-import { Link } from 'react-router-dom';
-
 /** icons **/
 import { BsPencilSquare, BsServer, BsDisplayFill, BsChevronRight, BsChevronLeft, BsCheck } from 'react-icons/bs'
+
 /** css **/
-import './Dashboard.css'
+import './Dashboard.css';
+
 // import DashboardModal from './DashboardModal'
-import homeTemplateImg1 from '../../images/home_page_template1.jpg'
-import homeTemplateImg2 from '../../images/home_page_template2.jpg'
+import homeTemplateImg1 from '../../images/home_page_template1.jpg';
+import homeTemplateImg2 from '../../images/home_page_template2.jpg';
 
-const Dashboard = () => {
+/** api to patch github token **/
+import api from '../../api/api';
 
-    const [show1, setShow1] = useState(false);
-    const [show2, setShow2] = useState(false);
-    const [show3, setShow3] = useState(false);
 
-    const handleCloseOne = () => setShow1(false);
-    const handleCloseTwo = () => setShow2(false);
-    const handleCloseThree = () => setShow3(false);
+/** Pass the github username and token into api **/
+const storeGithubToken = (teamId, githubUsername, githubToken) => {
+    // TODO: validate the input and token (token start with 'ghp_' and has total 40 characters)
+
+    try {
+        api.patch(`team/${teamId}/githubToken`, githubToken);
+        api.patch(`team/${teamId}/githubUsername`, githubUsername);
+
+    } catch (err) {
+        console.error(`Error in patching github token/username in Dashboard.js: ${err}`);
+    }
+
+}
+
+const Dashboard = () => {
 
-    const handleShowOne = () => setShow1(true);
-    const handleShowTwo = () => setShow2(true);
-    const handleShowThree = () => setShow3(true);
+    /** Managing the navigation between modals **/
+    const [githubModal, setShowGithub] = useState(false);
+    const [template1, setShowTemplate1] = useState(false);
+    const [template2, setShowTemplate2] = useState(false);
 
+    const closeGithubModal = () => setShowGithub(false);
+    const closeTemplate1 = () => setShowTemplate1(false);
+    const closeTemplate2 = () => setShowTemplate2(false);
 
+    const showGithubModal = () => setShowGithub(true);
+    const showTemplate1 = () => setShowTemplate1(true);
+    const showTemplate2 = () => setShowTemplate2(true);
+
+    /** Set the state of 'website.isCreated' to display different button**/
     const dispatch = useDispatch();
     const websiteIsCreated = useSelector(state => state.website.isCreated);

```

Comment:
Thunder has changed it so we now store a boolean under `state.team.repoCreated`, could we change this to that please?

Reply:
yes sure, we've changed it here, and will leave the backend reducer to `Thunder` (which means currently the button is not changing to `Eidt Website` after created because the state is not built in redux)

Reply:
Also, currently in `Dashboard.js` line 191 and line 223, there're two buttons that will need to trigger state change of the `repoCreated`, they are currently still calling `dispatch(createWebste())` which is the previous redux store i created (I've not removed it yet to avoid error), need to be updated once action for `state.team.repoCreated` is implemented in the branch by `Thunder`